pkg/versionstream/version_data.go

```diff
@@ -134,6 +134,12 @@ func verifyError(name string, err error) error {
 func convertToVersion(text string) string {
 	answer := strings.TrimSpace(text)
 	answer = strings.TrimPrefix(answer, "v")
+
+	// Git for Windows doesn't follow semver, example: 2.23.0.windows.1
+	// we're trimming here the ".windows.<number>" suffix from the currentVersion to avoid error described in issue #6825
+	r := regexp.MustCompile(`\.windows\.[0-9]+$`)
```

Comment:
I find this a bit to specific. _convertToVersion_ is really a generic function to extract a semantic version from the specified text. Adding this special case for how the git version looks seems a bit fishy imo.

What's about we turn this around and look via regular expression at the string _text_ and returns the match if it exists. If we want to keep the current function signature we can return the empty string if there is no match. In this case it will fail later on when we pass the returned string to `semver.Make`. Alternatively, we can change the signature to return the `string, error`.

Something like this might do 
```
regexp.Compile(`v?([0-9]+\.[0-9]+\.[0-9]+)`)
```